pxd.c

```diff
@@ -806,16 +780,37 @@ static int pxd_init_disk(struct pxd_device *pxd_dev, struct pxd_add_ext_out *add
 	  pxd_dev->tag_set.cmd_size = sizeof(struct fuse_req);
 
 	  err = blk_mq_alloc_tag_set(&pxd_dev->tag_set);
-	  if (err)
+	  if (err) {
 		goto out_disk;
+	  }
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,15,0)
+	  disk = blk_mq_alloc_disk(&pxd_dev->tag_set, pxd_dev);
+	  if (IS_ERR(disk)) {
+		err = PTR_ERR(disk);
```

Comment:
disk is initialized as a error pointer. This will get freed below when control jumps to out_disk.